src/client/options/parse.rs

```diff
@@ -80,6 +80,7 @@ impl Action for ParseConnectionString {
                                 "srvMaxHosts and loadBalanced=true cannot both be present",
                             ));
                         }
```

Comment:
I did a quick pass for any other `as usize` issues and added comments after I was certain they were safe